index.js

```diff
@@ -1,525 +1,559 @@
+				typeof options[key] === 'undefined' ||
+				options[key] === null
+			)
+				options[key] = clientOptions[key];
+		}
+		this.options = options;
+		this.data = {
+			messageCache: [],
+			bannedUsers: [],
+			kickedUsers: [],
+			warnedUsers: [],
+			mutedUsers: [],
+			users: []
+		};
+	}
+
+	/**
+	 * Checks a message.
+	 *
+	 * @param {Message} message The message to check.
+	 *
+	 * @returns {Promise<boolean>} Whether the message has triggered a threshold.
+	 *
+	 * @example
+	 * client.on('message', (msg) => {
+	 * 	antiSpam.message(msg);
+	 * });
+	 */
+	async message(message) {
+		const { options, data } = this;
+		if (
+			message.channel.type === 'dm' ||
+			message.author.id === message.client.user.id ||
+			(message.guild.ownerID === message.author.id && !options.debug)
+		)
+			return false;
+
+		if (version.split('.')[0] !== '12' && !message.member)
+			message.member = await (v11 ? message.guild.fetchMember(message.author) : message.guild.members.fetch(message.author));
+		if (
+			(options.ignoreBots && message.author.bot) ||
+			options.exemptPermissions.some(permission => message.member.hasPermission(permission))
+		)
+			return false;
+
+		if (
+			(v11 ? message.member.roles : message.member.roles.cache).some(role =>
+				typeof options.ignoredRoles === 'function'
+					? options.ignoredRoles(role)
+					: options.ignoredRoles.includes(role.id) || options.ignoredRoles.includes(role.name)
+			) ||
+			(typeof options.ignoredUsers === 'function'
+				? options.ignoredUsers(message.author)
+				: options.ignoredUsers.includes(message.author.id)) ||
+			(typeof options.ignoredGuilds === 'function'
+				? options.ignoredGuilds(message.guild)
+				: options.ignoredGuilds.includes(message.guild.id)) ||
+			(typeof options.ignoredChannels === 'function'
+				? options.ignoredChannels(message.channel)
+				: options.ignoredChannels.includes(message.channel.id))
+		)
+			return false;
+
+		const banUser = async () => {
+			data.messageCache = data.messageCache.filter(m => m.author !== message.author.id);
+			data.bannedUsers.push(message.author.id);
+
+			if (!message.member.bannable) {
+				if (options.verbose)
+					console.log(
+						`**${message.author.tag}** (ID: ${message.author.id}) could not be banned, insufficient permissions.`
+					);
+				if (options.errorMessages)
+					await message.channel
+						.send(format(options.banErrorMessage, message))
+						.catch((e) => {
+							if (options.verbose) console.error(e);
+						});
+				return false;
+			}
+
+			try {
+				const tempMember = message.member;
+				await message.member.ban({
+					reason: 'Spamming!',
+					days: options.deleteMessagesAfterBanForPastDays
+				});
+				if (options.banMessage)
+					await message.channel.send(format(options.banMessage, message)).catch(e => {
+						if (options.verbose) console.error(e);
+					});
+					if(options.modlogEnabled)
+					let modlogCh = message.guild.channels.cache.find(ch => ch.name == options.modLogChannel)

```

Comment:
Here we should support v11 of Discordjs. Also, we should be able to find the channel using ID. (these changes should be applied everywhere)
```suggestion
					const modlogCh = (v11 ? message.guild.channels : message.guild.channels.cache).find(ch => ch.name === options.modLogChannel || ch.id === options.modLogChannel)
```